scripts/lib/CIME/utils.py

```diff
@@ -41,7 +42,8 @@ def redirect_stdout_stderr(new_target):
 
 @contextmanager
 def redirect_logger(new_target, logger_name):
-
+    old_stdout = sys.stdout
+    sys.stdout = new_target
```

Comment:
Was this trying to fix some other issue? I have a logger fix coming. If you really want to redirect stdout, stderr, and logging, you can nest the context managers:

```
with redirect_logger(..):
    with redirect_stdout_stderr(..):
        ...
```